libs/proposals/src/lib/protocol-upgrade-proposals/use-time-to-upgrade.ts

```diff
@@ -7,6 +7,43 @@ const DEFAULT_POLLS = 10;
 const INTERVAL = 1000;
 const durations = [] as number[];
 
+/**
+ * Parses block duration value and output a number of milliseconds.
+ * @param input The block duration input from the API, e.g. 4m5.001s
+ * @returns A number of milliseconds
+ */
+export const parseDuration = (input: string) => {
+  // h  -> 60*60*1000
+  // m  -> 60*1000
+  // s  -> 1000
+  // ms -> 1
+  // µs -> 1/1000
+  // ns -> 1/1000000
+  let H = 0;
+  let M = 0;
+  let S = 0;
+  const lessThanSecond = /^[0-9.]+[nµm]*s$/gu.test(input);
+  const exp = /(?<hours>[0-9.]+h)?(?<minutes>[0-9.]+m)?(?<seconds>[0-9.]+s)?/gu;
+  const m = exp.exec(input);
+  const seconds = lessThanSecond ? input : m?.groups?.['seconds'];
```

Comment:
Suggestion (non-blocking): error handling would be a nice improvement here, perhaps throwing an error if none of the matching groups are present in the input and lessThanSecond is false